pkg/controller/release/scheduler.go

```diff
@@ -59,6 +60,19 @@ func (s *Scheduler) ScheduleRelease(rel *shipper.Release) (*releaseInfo, error)
 	}
 
 	releaseErrors := shippererrors.NewMultiError()
+	if rel.Spec.VirtualStrategy == nil {
+		strategy := rel.Spec.Environment.Strategy
+
+		surgeWeight, err := releaseutil.GetMaxSurgePercent(rel.Spec.Environment.Strategy.RollingUpdate, replicaCount)
+		if err != nil {
+			releaseErrors.Append(err)
+		}
+		virtualStrategy, err := releaseutil.BuildVirtualStrategy(strategy, surgeWeight)
```

Comment:
This should probably not execute if there is an error. Same goes for line 74 below.